pkg/controller/render/render_controller.go

```diff
@@ -636,3 +724,521 @@ func getMachineConfigsForPool(pool *mcfgv1.MachineConfigPool, configs []*mcfgv1.
+	// it was simple and straightforward, there's probably a podman/buildah/libcontainer equivalent
+	// that I should use instead.
+
+	// Get the ignition out of the MachineConfig
+	ignConf, err := ctrlcommon.ParseAndConvertConfig(generated.Spec.Config.Raw)
+	if err != nil {
+		glog.Errorf("parsing Ignition config failed with error: %v", err)
+	}
+
+	// Convert it to json
+	ignJSON, err := json.MarshalIndent(ignConf, "  ", "    ")
+	if err != nil {
+		glog.Warningf("Failed to marshal config for container: %s", err)
+	}
+
+	// Convert the MachineConfig to JSON also just in case we want it
+	machineConfigJSON, err := json.MarshalIndent(generated, "  ", "    ")
+	if err != nil {
+		glog.Warningf("Failed to marshal config for container: %s", err)
+	}
+
+	//TODO(jkyros): we could potentially expose the component machineconfigs or metadata here too
+
+	// This is the "file guts" for our "layer "
+	var fileMap = map[string][]byte{
+		"/machineconfig.json": machineConfigJSON,
+		"/ignition.json":      ignJSON,
+	}
+
+	// This is essentially "FROM scratch"
+	// ADD /machineconfig.json
+	// ADD /ignition.json
+	newImg, err := crane.Image(fileMap)
+	if err != nil {
+		glog.Warningf("Failed to create image layer: %s", err)
+	}
+
+	// Stuff a label in to see what happens when we build
+	newImg, err = mutate.Config(newImg, v1.Config{
+		Labels: map[string]string{"machineconfig": generated.Name},
+	})
+	if err != nil {
+		glog.Warningf("Failed to mutate image: %s", err)
+	}
+
+	// Tag it so we can push it into the integrated registry
+	// I added a RoleBinding manifest that gives us permission to push to the repo with our service account
+	// TODO(jkyros): centralize the naming of these streams
+	renderedImageStreamName := pool.Name + ctrlcommon.ImageStreamSuffixRenderedConfig
+	contentTag := path.Join("image-registry.openshift-image-registry.svc:5000", ctrlcommon.MCONamespace, renderedImageStreamName)
+	tag, err := name.NewTag(contentTag + ":latest")
+	if err != nil {
+		return fmt.Errorf("Failed to create new tag %s: %s", contentTag, err)
+	}
+
+	// The the trust for the repo serving cert is not in our system trust by default, but we know what signed it, so we
+	// get the pubkey for that and add it to our tls config so we trust it
+	serviceServingSignerCA, err := ctrl.kubeclient.CoreV1().ConfigMaps("openshift-config").Get(context.TODO(), "openshift-service-ca.crt", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve the openshift-service-ca.crt configmap: %w", err)
+	}
+
+	// TODO(jkyros): This is better than using skipverify/"nosec" to get past verify, but figure out error cases here later
+	rootCAs := x509.NewCertPool()
+	rootCAs.AppendCertsFromPEM([]byte(serviceServingSignerCA.Data["service-ca.crt"]))
+
+	// Add the service serving CA to our trust bundle in our tls config
+	tr := &http.Transport{TLSClientConfig: &tls.Config{
+		RootCAs: rootCAs,
+	}}
+
+	crane.WithTransport(tr)
```

Comment:
Do you need this here and passed to `crane.Push`?

Reply:
Nope, Good catch. I don't think I do. That's a no-op now. Leftovers from the code I threw away.